scipy/stats/biasedurn/mersenne.cpp

```diff
@@ -0,0 +1,183 @@
+/**************************   mersenne.cpp   **********************************
```

Comment:
This and `userintf.cpp` were the two files needed to get the code to run that we don't have explicit permission to use. Of course, we want to replace the functions from here with our own RNG. Question is - what is the best way to do that? We need to replace `RandomInit`, `Random`, and `IRandom`.

Reply:
A Mersenne Twister is available through numpy (see [here](https://numpy.org/doc/stable/reference/random/bit_generators/generated/numpy.random.MT19937.ctypes.html#numpy.random.MT19937.ctypes)).  Since it's available through ctypes, that means we should be able to link the library and use it

Reply:
I don't think we need Mersenne Twister in particular  - any uniformly distributed random numbers should do. I am most familiar with `np.random.seed/random/randint` but my understanding is that distributions are supposed to work with more modern random `Generator`, too. The trouble is that I don't know how to access Python things from C++, and the Cython documentation isn't as good about that as vice-versa. I will look up ctypes I guess.

Reply:
It may actually be even easier, C++11 has MT19937 built into the standard library.  I'm working wrapping it with the randomc.h interface

Reply:
@mckib2 please wait for @WarrenWeckesser to comment on that. I'm not sure that's what we want.

Reply:
It is pretty much a requirement that the underlying random number generator must be numpy.  That's part of the API of the `rvs` methods of the distributions in scipy.stats.

Reply:
Didn't see this in time, but it was easy enough to do (see [here](https://github.com/mdhaber/scipy/pull/32/files)).  That removes mersenne.cpp  and randomc.h from the build, but I'll let Warren comment on whether is this is a good solution or not